src/core/features/course/pages/list-mod-type/list-mod-type.ts

```diff
@@ -125,19 +141,64 @@ export class CoreCourseListModTypePage implements OnInit {
             const result = await CoreCourseHelper.addHandlerDataForModules(sections, this.courseId);
 
             this.sections = result.sections;
+
+            this.canLoadMore = false;
+            this.showSectionId = 0;
+            this.showMoreActivities();
         } catch (error) {
             CoreDomUtils.showErrorModalDefault(error, 'Error getting data');
         }
     }
 
+    /**
+     * Show more activities (only used when showing all the sections at the same time).
+     *
+     * @param infiniteComplete Infinite scroll complete function. Only used from core-infinite-loading.
+     */
+    showMoreActivities(infiniteComplete?: () => void): void {
+        this.canLoadMore = false;
+
+        const sections = this.sections || [];
+        let modulesLoaded = 0;
+        let i: number;
+        for (i = this.showSectionId + 1; i < sections.length; i++) {
+            if (!sections[i].hasContent || !sections[i].modules) {
+                continue;
+            }
+
+            modulesLoaded += sections[i].modules.length;
+
+            if (modulesLoaded >= CoreCourseListModTypePage.LOAD_MORE_ACTIVITIES) {
+                break;
+            }
+        }
+
+        this.showSectionId = i;
+        this.canLoadMore = i < sections.length;
```

Comment:
```suggestion
        this.canLoadMore = i < sections.length - 1;
```
Shouldn't this code be like this? If I'm not wrong, if `i` is `sections.length - 1` it means that the last section was already treated so there are no more to treat. Your code handles it properly in the next if, but IMO there's no need to in this case.